### PR TITLE
Harmonize FileInput component with styleguide

### PR DESCRIPTION
**Pull request by fscholdei**
## Proposed changes

Harmonize FileInput component with styleguide

### Your checklist for this pull request

**Contributor**:
- [x] PR is well described and the description can be used as a commit message on squash
- [x] Related issues linked to PR if existing and labels set
- [x] Target branch is not master (in most cases develop should bet the target of choice) 
- [x] Code does not conflict with target branch
- [ ] New code is covered with unit tests
- [x] New ui components are tested inside the storybook (module ui-components only) 
- [x] [Changelog entry file](https://github.com/scm-manager/changelog#changelog-entry-files) created in `gradle/changelog` or CHANGELOG.md is updated for plugins
- [ ] Documentation updated (only necessary for new features or changed behaviour)

**Reviewer**:
- [ ] The clean code principles are respected ([CleanCode](https://clean-code-developer.com/virtues/))
- [ ] All new code/logic is implemented on the right spot / "Should this be done here?"
- [ ] UI changes fits into the layout
- [ ] The UI views / components are responsive (mobile views)
- [ ] Correct translations are available

### Checklist for branch merge request (not required for forks)

- [ ] Branch is green/blue on [Jenkins](https://oss.cloudogu.com/jenkins/)
- [ ] Quality Gate passed on [SonarQube](https://sonarcloud.io/organizations/scm-manager/projects)

### CODE REVIEW

File: scm-ui/ui-components/src/forms/FileInput.tsx [hunk 1]
```diff
@@ -66,18 +75,31 @@ const FileInput: FC<Props> = ({
...
+          <span className="file-cta">
+            <span className="file-icon">
+              <i className="fas fa-arrow-circle-up" />
+            </span>
+            <span className="file-label has-text-weight-bold">{t("fileInput.label")}</span>
+          </span>
+          {file?.name ? (
+            <span className="file-name">{file?.name}</span>
+          ) : (
+            <span className="file-name has-text-weight-light has-text-grey-light">{t("fileInput.noFileChosen")}</span>
```
Comment (eheimbuch) [inline scm-ui/ui-components/src/forms/FileInput.tsx:100]: Before we supported custom filename placeholder. This is useful to provide file restrictions like we did in the repository-avatar-plugin.

```suggestion
            <span className="file-name has-text-weight-light has-text-grey-light">{filenamePlaceholder || t("fileInput.noFileChosen")}</span>
```

File: scm-ui/ui-components/src/forms/FileInput.tsx [hunk 2]
```diff
@@ -49,9 +51,16 @@ const FileInput: FC<Props> = ({
...
-  onChange
+  onChange,
 }) => {
+  const [t] = useTranslation("commons");
+  const [file, setFile] = useState<File | null>(null);
+
   const handleChange = (event: ChangeEvent<HTMLInputElement>) => {
+    const uploadedFile = event?.target?.files![0];
+    // @ts-ignore the uploaded file doesn't match our types
+    setFile(uploadedFile);
```
Comment (eheimbuch) [inline scm-ui/ui-components/src/forms/FileInput.tsx:62]: You copied this from the FileUpload component. Do we really need this? The filename was also shown before correctly. I think we should look for a better solution. You may ask @Pilopa about this.

**Review by eheimbuch — CHANGES_REQUESTED** [review]

File: scm-ui/ui-components/src/forms/FileInput.tsx (hunk 1 above)
Comment (fscholdei) [reply scm-ui/ui-components/src/forms/FileInput.tsx:100]: Only the FileUpload component supported this feature. FileInput did not. I can add it for future use 👍

File: scm-ui/ui-components/src/forms/FileInput.tsx (hunk 2 above)
Comment (fscholdei) [reply scm-ui/ui-components/src/forms/FileInput.tsx:62]: Good question. Due to the custom styling, the input itself is no longer visible and only a span.file-name is available in which the file name must be set somehow...

Comment (Pilopa) [reply scm-ui/ui-components/src/forms/FileInput.tsx:64]: @eheimbuch Its okay like this. There is unfortunately no better solution that I am aware of after spending some time researching the issue.

**Pull request merged**